images/ci/Dockerfile

```
@@ -0,0 +1,33 @@
+# Build the external-secrets-operator binary
+FROM docker.io/golang:1.23 AS builder
+ARG TARGETOS
+ARG TARGETARCH
+
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# Copy the go source
+COPY cmd/external-secrets-operator/main.go cmd/external-secrets-operator/main.go
+COPY api/ api/
+COPY pkg/controller/ pkg/controller/
+
+# Build
+# the GOARCH has not a default value to allow the binary be built according to the host where the command
+# was called. For example, if we call make docker-build in a local env which has the Apple Silicon M1 SO
+# the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
+# by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
+RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o external-secrets-operator cmd/external-secrets-operator/main.go
```

```suggestion
ENV CGO_ENABLED=1

RUN GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o external-secrets-operator cmd/external-secrets-operator/main.go
```

We need CGO_ENABLED for FIPS.